src/main/java/org/cyclops/integrateddynamics/core/evaluate/variable/ValueParseOperators.java

```diff
@@ -0,0 +1,97 @@
+                } else {
+                    return ValueTypeBoolean.ValueBoolean.of(true);
+                }
+            } catch (NumberFormatException e) {
+                return ValueTypeBoolean.ValueBoolean.of(true);
+            }
+        }));
+
+    public static IOperator PARSE_DOUBLE = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.DOUBLE, v -> {
+            ValueTypeString.ValueString value = v.getValue(0);
+            // TODO: Floating point Hex/Octal
+            try {
+                return ValueTypeDouble.ValueDouble.of(Double.parseDouble(value.getRawValue()));
+            } catch (NumberFormatException e) {
+                try {
+                    // \u221E = infinity symbol
+                    Pattern p = Pattern.compile("\\A([+-]?)(Inf(inity)?|\u221E)\\z", Pattern.CASE_INSENSITIVE);
+                    Matcher m = p.matcher(value.getRawValue().trim());
+                    if(m.matches()){
+                        if(m.group(1).equals("-")){
+                            return ValueTypeDouble.ValueDouble.of(Double.NEGATIVE_INFINITY);
+                        }
+                        return ValueTypeDouble.ValueDouble.of(Double.POSITIVE_INFINITY);
+                    }
+                    // Try as a long
+                    return ValueTypeDouble.ValueDouble.of((double) Long.decode(value.getRawValue()));
+                } catch (NumberFormatException e2) {
+                    return ValueTypeDouble.ValueDouble.of(0.0);
+                }
+            }
+        }));
+
+    public static IOperator PARSE_INTEGER = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.INTEGER, v -> {
+            ValueTypeString.ValueString value = v.getValue(0);
+            try{
+                return ValueTypeInteger.ValueInteger.of(Integer.decode(value.getRawValue()));
+            } catch (NumberFormatException e) {
+                return ValueTypeInteger.ValueInteger.of(0);
+            }
+        }));
+
+    public static IOperator PARSE_LONG = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.LONG, v -> {
+            ValueTypeString.ValueString value = v.getValue(0);
+            try {
+                return ValueTypeLong.ValueLong.of(Long.decode(value.getRawValue()));
+            } catch (NumberFormatException e) {
+                return ValueTypeLong.ValueLong.of(0L);
+            }
+        }));
+
+    public static IOperator PARSE_NBT = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.NBT, v -> {
+            ValueTypeString.ValueString value = v.getValue(0);
+            try {
+                return new ValueTypeNbt().deserialize(value.getRawValue());
+            } catch (IllegalArgumentException e) {
+                return ValueTypeNbt.ValueNbt.of(null);
+            }
+        }));
+
+    public static IOperator PARSE_STRING = Operators.REGISTRY.register(new ParseOperator<>(ValueTypes.STRING, value -> value.getValue(0)));
```

Comment:
I suggest removing this one.

Reply:
Unneeded? Yes. Symmetrical with what one would expect? `(int) (int) 0` Probably. Either way it's gone now.